File: backend/src/main/java/com/radar/prospection/scheduler/RadarScheduler.java
package com.radar.prospection.scheduler;

import com.radar.prospection.scraper.ScraperCoordinator;
import com.radar.prospection.signal.SignalCoordinator;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Slf4j
@Component
@RequiredArgsConstructor
public class RadarScheduler {

    private final ScraperCoordinator scraperCoordinator;
    private final SignalCoordinator signalCoordinator;

    // 3x par jour : 8h, 12h, 18h
    @Scheduled(cron = "${radar.scheduling.job-boards-cron}")
    public void scanJobBoards() {
        log.info("⏰ Scan job boards déclenché");
        ScraperCoordinator.ScrapingResult result = scraperCoordinator.runAll();
        log.info("✅ Job boards : {} nouvelles missions (score fit calculé par Claude)", result.newMissions());
    }

    // 1x par jour à 8h30
    @Scheduled(cron = "${radar.scheduling.signals-cron}")
    public void scanSignals() {
        log.info("⏰ Scan signaux marché caché déclenché");
        SignalCoordinator.SignalScanResult result = signalCoordinator.runAll();
        log.info("✅ Signaux : {} nouveaux signaux détectés", result.newSignals());
    }
}

File: backend/src/main/kotlin/com/radar/prospection/scheduler/RadarScheduler.kt
package com.radar.prospection.scheduler

import com.radar.prospection.scraper.ScraperCoordinator
import com.radar.prospection.signal.SignalCoordinator
import org.slf4j.LoggerFactory
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class RadarScheduler(
    private val scraperCoordinator: ScraperCoordinator,
    private val signalCoordinator: SignalCoordinator
) {
    private val log = LoggerFactory.getLogger(javaClass)

    @Scheduled(cron = "\${radar.scheduling.job-boards-cron}")
    fun scanJobBoards() {
        log.info("Scan job boards déclenché")
        val result = scraperCoordinator.runAll()
        log.info("Job boards : {} nouvelles missions", result.newMissions)
    }

    @Scheduled(cron = "\${radar.scheduling.signals-cron}")
    fun scanSignals() {
        log.info("Scan signaux marché caché déclenché")
        val result = signalCoordinator.runAll()
        log.info("Signaux : {} nouveaux signaux détectés", result.newSignals)
    }
}
